### PR TITLE
Introduce subtext to block item

### LINKED ISSUE #8609: Add the best views date to AllTime block in Stats
In the All-time Stats in Insights there should be a new line under the "Best views ever" which tells the user the date when this happened.

### PR DESCRIPTION
**Pull request by planarvoid**
Fixes #8609 
This PR is adding the "Best views date" to the "All time insight" block under the "Best Views Ever" item.
I'm also introducing Constraint layout for the default block list item. The original layout was getting more complex and this should make it easier for further development.

To test:
* Open Stats
* Scroll to "All Time Stats"
* The line "Best Views Ever" now contains a date under the text

### CODE REVIEW

**Comment by malinajirka** [comment]
Thanks @planarvoid! Looks good, I've left a few comments. I've also noticed author/follower icons in Comments/Followers section disappear after rotation.

**Comment by malinajirka** [comment]
Btw we might want to consider creating an issue for extracting the dimensions into a dimens.xml. Wdyt?

**Comment by planarvoid** [comment]
@malinajirka good point, I've created an issue here #8671

File: WordPress/src/test/java/org/wordpress/android/ui/stats/refresh/StatsDateFormatterTest.kt [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import com.nhaarman.mockito_kotlin.whenever
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Before
+import org.junit.Test
+import org.mockito.Mock
+import org.wordpress.android.BaseUnitTest
+import org.wordpress.android.util.LocaleManagerWrapper
+import java.util.Locale
+
+class StatsDateFormatterTest : BaseUnitTest() {
```
[the remark this answers is not in the document]
Comment (malinajirka) [reply WordPress/src/test/java/org/wordpress/android/ui/stats/refresh/StatsDateFormatterTest.kt]: I don't think so. We are not directly calling SimpleDateFormat with the locale -> we are calling StatsDateFormatter which may theoretically ignore the localeManagerWrapper and use SimpleDateFormat with a hardcoded locale.

**Comment by malinajirka** [comment]
I don't want to block this PR on the minor issue/suggestion. Feel free to fix it if you want in another PR.

**Pull request merged**